torch/_dynamo/variables/builtin.py

```diff
@@ -1465,12 +1476,16 @@ def _unimplemented():
         if isinstance(left, UserDefinedObjectVariable) and isinstance(
             right, UserDefinedObjectVariable
         ):
-            return ConstantVariable.create(op(left.value, right.value))
-
-        if op.__name__ == "is_":
-            # If the two objects are of different type, we can safely return False
-            if type(left) is not type(right):
-                return ConstantVariable.create(False)
+            ret = op(left.value, right.value)
+            if op is operator.is_ and not ret:
+                guard = left.source.make_guard(
+                    functools.partial(
+                        GuardBuilder.UNALIASED_INPUT, source_b=right.source
+                    )
+                )
+                return ConstantVariable.create(ret, guards=set({guard}))
+            else:
+                return ConstantVariable.create(ret)
```

Comment:
I think you need the inverse guard in this case

Reply:
Actually the guard is unnecessary as we deduplicate VTs and install guards when there is aliasing.